bindings/pyroot/pythonizations/python/ROOT/_pythonization/_rdataframe.py

```diff
@@ -8,6 +8,164 @@
+
+~~~{.py}
+@ROOT.Numba.Declare(['RVec<float>', 'int'], 'RVec<float>')
+def pypowarray(x, y):
+    return x**y
+
+df.Define('array', 'ROOT::RVec<float>{1.,2.,3.})\
+  .Define('arraySquared', 'Numba::pypowarray(array, 2)')
+~~~
+
+Note that this functionality requires the Python packages `numba` and `cffi` to be installed.
+
+### Interoperability with NumPy
+
+#### Conversion to NumPy arrays
+
+Eventually, you probably would like to inspect the content of the RDataFrame or process the data further
+with Python libraries. For this purpose, we provide the `AsNumpy()` function, which returns the columns
+of your RDataFrame as a dictionary of NumPy arrays. See a simple example below or a full tutorial [here](df026__AsNumpyArrays_8py.html).
+
+~~~{.py}
+df = ROOT.RDataFrame("myTree", "myFile.root")
+cols = df.Filter("x > 10").AsNumpy(["x", "y"]) # retrieve columns "x" and "y" as NumPy arrays
+print(cols["x"], cols["y"]) # the values of the cols dictionary are NumPy arrays
+~~~
+
+#### Processing data stored in NumPy arrays
+
+In case you have data in NumPy arrays in Python and you want to process the data with ROOT, you can easily
+create an RDataFrame using `ROOT.RDF.MakeNumpyDataFrame`. The factory function accepts a dictionary where
+the keys are the column names and the values are NumPy arrays, and returns a new RDataFrame with the provided
+columns.
+
+Only arrays of fundamental types (integers and floating point values) are supported and the arrays must have the same length.
+Data is read directly from the arrays: no copies are performed.
+
+~~~{.py}
+# Read data from NumPy arrays
+# The column names in the RDataFrame are taken from the dictionary keys
+x, y = numpy.array([1, 2, 3]), numpy.array([4, 5, 6])
+df = ROOT.RDF.MakeNumpyDataFrame({"x": x, "y": y})
+
+# Use RDataFrame as usual, e.g. write out a ROOT file
+df.Define("z", "x + y").Snapshot("tree", "file.root")
+~~~
+
+### Construct histogram and profile models from a tuple
+
+The Histo1D(), Histo2D(), Histo3D(), Profile1D() and Profile2D() methods return
+histograms and profiles, respectively, which can be constructed using a model
+argument.
+
+In Python, we can specify the arguments for the constructor of such histogram or
+profile model with a Python tuple, as shown in the example below:
+
+~~~{.py}
+# First argument is a tuple with the arguments to construct a TH1D model
+h = df.Histo1D(("histName", "histTitle", 64, 0., 128.), "myColumn")
+~~~
+
+### AsRNode helper function
+
+The ROOT::RDF::AsRNode function casts an RDataFrame node to the generic ROOT::RDF::RNode type. From Python, it can be used to pass any RDataFrame node as an argument of a C++ function, as shown below:
+
+~~~{.py}
+ROOT.gInterpreter.Declare("""
+ROOT::RDF::RNode MyTransformation(ROOT::RDF::RNode &df) {
+    auto myFunc = [](float x){ return -x;};
+    return df.Define("y", myFunc, {"x"});
+}
+""")
+
+df = ROOT.RDataFrame("myTree", "myFile.root")
+df = ROOT.MyTransformation(ROOT.RDF.AsRNode(df))
```

Comment:
I would add a further example that shows that other nodes can be casted, e.g.
```python
node = df.Filter("x > 42")
node = ROOT.MyTransformation(ROOT.RDF.AsRNode(node))
```

Reply:
Done!